src/toil/utils/toilDebugJob.py

```diff
@@ -23,28 +27,50 @@
 
 
 def main() -> None:
-    parser = parser_with_common_options(jobstore_option=True)
-    parser.add_argument("jobID", nargs=1,
+    parser = parser_with_common_options(jobstore_option=True, prog="toil debug-job")
+    parser.add_argument("jobID", type=str, nargs='?', default=None,
                         help="The job store id of a job within the provided jobstore to run by itself.")
-    parser.add_argument("--printJobInfo", nargs=1,
-                        help="Return information about this job to the user including preceding jobs, "
-                             "inputs, outputs, and runtime from the last known run.")
+    parser.add_argument("--printJobInfo", type=str,
+                        help="Dump debugging info about this job ID")
 
     options = parser.parse_args()
     set_logging_from_options(options)
-    config = Config()
+
+    jobStore = Toil.resumeJobStore(options.jobStore)
+    # Get the config with the workflow ID from the job store
+    config = jobStore.config
+    # But override its options
     config.setOptions(options)
 
-    jobStore = Toil.resumeJobStore(config.jobStore)
+    did_something = False
 
     if options.printJobInfo:
-        printContentsOfJobStore(jobStorePath=config.jobStore, nameOfJob=options.printJobInfo)
+        if isinstance(jobStore, FileJobStore):
+            # List all its files if we can
+            printContentsOfJobStore(job_store=jobStore, job_id=options.printJobInfo)
+        # Print the job description itself
+        job_desc = jobStore.load_job(options.printJobInfo)
+        print(f"Job: {job_desc}")
+        pprint.pprint(job_desc.__dict__)
+
+        did_something = True
 
     # TODO: Option to print list of successor jobs
     # TODO: Option to run job within python debugger, allowing step through of arguments
     # idea would be to have option to import pdb and set breakpoint at the start of the user's code
 
-    jobID = options.jobID[0]
-    logger.debug(f"Running the following job locally: {jobID}")
-    workerScript(jobStore, config, jobID, jobID, redirectOutputToLogFile=False)
-    logger.debug(f"Finished running: {jobID}")
+    if options.jobID is not None:
+        # We actually want to run a job.
+
+        jobID = options.jobID
+        logger.debug(f"Running the following job locally: {jobID}")
+        workerScript(jobStore, config, jobID, jobID, redirectOutputToLogFile=False)
+        logger.debug(f"Finished running: {jobID}")
+        # Even if the job fails, the worker script succeeds unless something goes wrong with it internally.
+
+        did_something = True
+
+    if not did_something:
+        # Somebody forgot to tell us to do anything.
+        logger.error("Not doing anything. Provide a job ID to run, or an option.")
+        sys.exit(1)
```

Comment:
Should this error?  I'd expect `--help` to pop up (and maybe a log warning?) if the user didn't include any arguments with the command.